### PR TITLE
Dns

### PR DESCRIPTION
**Pull request by silenius**
### What does this PR do?

The parsing logic of the resolv.conf file is currently done in grains/core.py. This pull request move the logic in a dedicated salt/utils/dns.py module. It could be reused in other modules, like modules/debian_ip.py, etc 
### What issues does this PR fix or reference?

none
### Tests written?

No

### CODE REVIEW

**Comment by cachedout** [comment]
Hi @silenius 

Could you please address the lint issues here? Thanks. https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/2332/violations/

File: salt/utils/dns.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+
+import itertools
```
Comment (cachedout) [inline salt/utils/dns.py]: This needs a `  # pylint: disable=E0403` after it to disable this pylint check, please.

**Comment by silenius** [comment]
it should be OK :)

**Comment by damon-atkins** [comment]
Would it not be best to move it into salt.modules.network and join the likes of salt.modules.network.dig , salt.modules.network.get_hostname , etc.

**Comment by cachedout** [comment]
@damon-atkins I don't think so. We have to be careful not to create circular dependencies between grains and execution modules. I think that utils is the right place for this code and if we want to increase re-use, an execution module could be made to call out to it from there.

**Comment by silenius** [comment]
maybe it could be moved in the existing `salt.utils.network` module ...

**Comment by cachedout** [comment]
@silenius That's sensible. Do you want to move it there before I merge this?

**Comment by silenius** [comment]
for me it could stay in a separate `salt.utils.dns` module, and can always be imported in `salt.utils.network` at later stage ..

**Comment by cachedout** [comment]
@silenius OK. I'll get this in then. Thanks!

**Pull request merged**